test.js

```diff
@@ -0,0 +1,124 @@
+const app = require('express')()
+const bodyParser = require('body-parser')
+const multer = require('multer') // v1.0.5
+const upload = multer() // for parsing multipart/form-data
+
+app.use(bodyParser.json()) // for parsing application/json
+app.use(bodyParser.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
+
+app.post('/profile', upload.array(), function (req, res, next) {
+  console.log(req.body)
+  res.json(req.body)
+})
+
+const port = 3001;
+
+app.get('/', (req, res) => res.send('Hello World!'));
+
+
+class Treenode {
+
+  constructor(label) {
+    this.id = (++Treenode.id_gen).toString()
+    // Treenode.lineage[this.id]=[label]
+    this.label =  label
+    this.parent_id = null
+    // this.parent_label = null
+    this.children = []
+    // Treenode.label_id[this.id] = label
+  }
+
+  add_child(child){
+    this.children.push(child)
+    child.parent_id = this.id
+  }
+
+}
+
+function find_id(iter, parent_id) {
+  if (iter.id === parent_id) {
+    return iter
+  }
+  else {
+    if (iter.children.length > 0) {
+      for (var i = 0; i < iter.children.length; i++){
```

Comment:
in terms of optimization, this at most would be a linear search through all nodes under a parent to find an id O(n). 
If we consider the manager, it can do things like maintain a mapping of all nodes ever existing and have a direct link to it making lookups in constant time O(1)
```
const nodeMap = {
  1: TreeNode(),
  2: TreeNode(),
  3: TreeNode(),
}
```

Reply:
OMG!!!! I WAS TRYING TO FIGURE HOW TO DO THIS IN JS!!!!
THANK YOU!!!! I spent an hour trying to figure it out